### PR TITLE
Create test_nestedattr.py

### PR DESCRIPTION
**Pull request by cenouralm**

### CODE REVIEW

File: tests/test_nestedattr.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @property
+    def artist(self):
+        return {'name': 'property'}
+
+
+album = MyAlbum(
+    artist={'name': 'david'},  # this shouldn't be used by the schema
+    title="Hunky Dory",
+    release_date=date(1971, 12, 17)
+)
```
Comment (ppanero) [inline tests/test_nestedattr.py:32]: moderate: is this used?

File: tests/test_nestedattr.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from datetime import date
+from pprint import pprint
```
Comment (ppanero) [inline tests/test_nestedattr.py:2]: major: unused imports

File: tests/test_nestedattr.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+album = MyAlbum(
+    artist={'name': 'david'},  # this shouldn't be used by the schema
+    title="Hunky Dory",
+    release_date=date(1971, 12, 17)
+)
+
+
+def test_test_schema():
```
Comment (ppanero) [inline tests/test_nestedattr.py:35]: moderate: naming

File: tests/test_nestedattr.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    album = MyAlbum(
+        title="Hunky Dory",
+        release_date=date(1971, 12, 17)
```
Comment (ppanero) [inline tests/test_nestedattr.py:38]: doubt: below it dumps this object which *doesn't have* any `artist`... How does it actually work then? Maybe I'm not understanding the usage of this field. @slint Can confirm.

Comment (slint) [reply tests/test_nestedattr.py:38]: I suggest adding something like this also:

```python
album = MyAlbum(
    artist={'name': 'david'},  # this shouldn't be used in the schema dump
    title="Hunky Dory",
    release_date=date(1971, 12, 17)
)
```

Maybe then the `artist` property could be something like:

```python
class MyAlbum(dict):

    @property
    def artist(self):
        return {'name': self['artist']['name'].catpialize()}
```

to showcase that we prefer the property/attribute's behavior compared to `__getitem__` from the dict

File: tests/test_nestedattr.py (hunk 2 above)
Comment (cenouralm) [reply tests/test_nestedattr.py:2]: `from datetime import date` is used twice on `release_date=date(1971, 12, 17)`

File: tests/test_nestedattr.py (hunk 1 above)
Comment (cenouralm) [reply tests/test_nestedattr.py:32]: Not anymore!

File: tests/test_nestedattr.py (hunk 2 above)
Comment (ppanero) [reply tests/test_nestedattr.py:2]: meant `pprint` which is the commented line :)

**Pull request merged**